src/lgdo/lh5/iterator.py

```diff
@@ -676,3 +697,390 @@ def __next__(self) -> Table:
+
+              - ``NDArray``: if 1D list of values; if 2D list of lists of values in
+                same order as axes
+              - ``Collection[ArrayLike]``: return list of values in same order as axes
+              - ``Mapping[str, ArrayLike]``: mapping from axis name to values
+              - ``pandas.DataFrame``: pandas dataframe. Treat as mapping from column
+                name to values
+
+            - A string expression. This will call `pd.DataFrame.query <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.query.html>`_ and return
+              a pandas DataFrame containing all columns in the fields mask.
+
+        processes:
+            number of processes. If ``None``, use number equal to threads available
+            to ``executor`` (if provided), or else do not parallelize
+        executor:
+            `concurrent.futures.Executor <https://docs.python.org/3/library/concurrent.futures.html>`_
+            object for managing parallelism. If ``None``, create a ``ProcessPoolExecutor`` with
+            number of processes equal to ``processes``.
+        """
+        if where is None:
+            where = _identity
+
+        if isinstance(where, str):
+            where = _pandas_query(where)
+
+        test = where(self.lh5_buffer, self)
+        if isinstance(test, LGDOCollection):
+            it = self.map(
+                where, processes=processes, executor=executor, aggregate=Table.append
+            )
+            if isinstance(it, LGDOCollection):
+                return it
+            ret = next(it)
+            for res in it:
+                ret.append(res)
+            return ret
+        if isinstance(test, pd.DataFrame):
+            return pd.concat(
+                list(self.map(where, processes=processes, executor=executor)),
+                ignore_index=True,
+            )
+        if isinstance(test, np.ndarray):
+            return np.concatenate(
+                list(self.map(where, processes=processes, executor=executor))
+            )
+        if isinstance(test, ak.Array):
+            return ak.concatenate(
+                list(self.map(where, processes=processes, executor=executor))
+            )
+
+        msg = f"Cannot call query with return type {test.__class__}. "
+        "Allowed return types: LGDOCollection, np.array, pd.DataFrame, ak.Array"
+        raise ValueError(msg)
+
+    def hist(
+        self,
+        ax: Hist | axis | Collection[axis],
+        where: Callable | str = None,
+        keys: Collection[str] | str = None,
+        processes: Executor | int = None,
+        executor: Executor = None,
+        **hist_kwargs,
+    ) -> Hist:
+        """
+        Fill a histogram from data produced by our ``where``. If
+        ``where`` is ``None``, fill with all data fetched by iterator.
+
+        Parameters
+        ----------
+        ax:
+            Axis object(s) used to construct the histogram. Can provide a ``Hist``
```

Comment:
```suggestion
            Axis object(s) used to construct the histogram. Can provide a :class:``.types.hist.Hist`
```
or whatever is the right path